__tests__/app.test.js

```diff
@@ -402,4 +402,125 @@ describe ("DELETE /api/comments/:comment_id", () => {
+                expect(article).toHaveProperty("article_img_url", expect.any(String));
+                expect(article).toHaveProperty("comment_count", expect.any(Number));
+                expect(article).not.toHaveProperty("body"); 
+            })        
+        })
+      }); 
+      test ("404:should return an error respond when topic is valid,but does not exist", () => {
+         return request(app)
+            .get("/api/articles?topic=notexist")
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("Not found");
+            });
+       });
+
+      // Check default sort by date and specified ASC order
+      test ('200:accepts a sort_by query with default sorts by created_at  and ASC order)',() => {
+        return request(app)
+        .get("/api/articles?order=ASC")
+        .expect(200)
+        .then(({body}) =>{
+         const {articles} = body;
+         expect(articles).toHaveLength(5);
+         expect(articles).toBeSortedBy("created_at",{ascending: true,});
+        })
+       })
+
+    // Sort_by any valid column and specified order   
+    test ('200:accepts a sort_by query with sorts by article_id and DESC order)',() => {
+        return request(app)
+        .get("/api/articles?sort_by=article_id&order=DESC")
+        .expect(200)
+        .then(({body}) =>{
+         const {articles} = body;
+         expect(articles).toHaveLength(5);
+         expect(articles).toBeSortedBy("article_id",{descending: true,});
+        })
+   })
+   test ('200:accepts a sort_by query with sorts by title and ASC order)',() => {
+    return request(app)
+    .get("/api/articles?sort_by=title&order=ASC")
+    .expect(200)
+    .then(({body}) =>{
+     const {articles} = body;
+     expect(articles).toHaveLength(5);
+     expect(articles).toBeSortedBy("title",{ascending: true,});
+    })
+   })
+    test ('200:accepts a sort_by query with sorts by author and ASC order)',() => {
+        return request(app)
+        .get("/api/articles?sort_by=author&order=ASC")
+        .expect(200)
+        .then(({body}) =>{
+         const {articles} = body;
+         expect(articles).toHaveLength(5);
+         expect(articles).toBeSortedBy("author",{ascending: true,});
+        })
+   })
+
+   test ('200:accepts a sort_by query with sorts by votes and default order)',() => {
+    return request(app)
+    .get("/api/articles?sort_by=votes")
+    .expect(200)
+    .then(({body}) =>{
+     const {articles} = body;
+     expect(articles).toHaveLength(5);
+     expect(articles).toBeSortedBy("votes",{descending: true,});
+    })
+   })
+   
+   test ('200:accepts a sort_by query with sorts by article_img_url and ASC order)',() => {
+    return request(app)
+     .get("/api/articles?sort_by=article_img_url&order=ASC")
+     .expect(200)
+     .then(({body}) =>{
+     const {articles} = body;
+     expect(articles).toHaveLength(5);
+     expect(articles).toBeSortedBy("article_img_url",{ascending: true,});
+    })
```

Comment:
These tests are very thorough and great, however, I don't think we need to test that we can sort by every possible property. Just one is fine 👍 😄

Reply:
Deleted